src/os/bluestore/BlueStore.cc

```diff
@@ -8988,6 +8988,55 @@ void BlueStore::trim_free_space(const string& type, std::ostream& outss)
   }
 }
 
+int BlueStore::zap_device(CephContext* cct, const string& dev, uint64_t gap_size)
+{
+  string path; // dummy var for dout
+  dout(5)<<  __func__ << " " << dev << dendl;
+  auto * _bdev =
+    BlockDevice::create(cct, dev, nullptr, nullptr, nullptr, nullptr);
+  int r = _bdev->open(dev);
+  if (r < 0)
+    goto fail;
+  if (!gap_size) {
```

Comment:
I think any freedom around selecting zap brush size (gap_size) is redundant.
We want to brush just and exactly as much as needed, which is:
`max(BDEV_LABEL_BLOCK_SIZE, bdev->get_block_size())`

My proposal is to fix it.

Reply:
For what it's worth, this is only my opinion, but what I expect from the `zap-device` command is simply to remove the Bluestore signature and nothing else.
I wouldn't really care about the block size being written, as long as the Bluestore signatures are removed altogether. Your call.